llarp/CMakeLists.txt

```diff
@@ -242,65 +324,206 @@ add_library(lokinet-amalgum
   service/tag.cpp
 )
 
-set(BOOTSTRAP_FALLBACKS)
-foreach(bs IN ITEMS MAINNET TESTNET)
-  if(BOOTSTRAP_FALLBACK_${bs})
-    message(STATUS "Building with ${bs} fallback boostrap path \"${BOOTSTRAP_FALLBACK_${bs}}\"")
-    file(READ "${BOOTSTRAP_FALLBACK_${bs}}" bs_data HEX)
-    if(bs STREQUAL TESTNET)
-      set(network "gamma")
-    elseif(bs STREQUAL MAINNET)
-      set(network "lokinet")
-    else()
-      string(TOLOWER "${bs}" network)
-    endif()
-    string(REGEX REPLACE "([0-9a-f][0-9a-f])" "\\\\x\\1" bs_data "${bs_data}")
-    set(BOOTSTRAP_FALLBACKS "${BOOTSTRAP_FALLBACKS}{\"${network}\"s, \"${bs_data}\"sv},\n")
-  endif()
-endforeach()
-configure_file("bootstrap-fallbacks.cpp.in" "${CMAKE_CURRENT_BINARY_DIR}/bootstrap-fallbacks.cpp" @ONLY)
-target_sources(lokinet-amalgum PRIVATE "${CMAKE_CURRENT_BINARY_DIR}/bootstrap-fallbacks.cpp")
+add_library(lokinet-layer-platform
+  STATIC
+  layers/platform/stub.cpp # todo: remove me
+)
+
 
-if(WITH_PEERSTATS_BACKEND)
-  target_compile_definitions(lokinet-amalgum PRIVATE -DLOKINET_PEERSTATS_BACKEND)
-  target_link_libraries(lokinet-amalgum PUBLIC sqlite_orm)
-endif()
 
+
+# interal tooling for pybind
+add_library(lokinet-tooling INTERFACE)
 if(WITH_HIVE)
-  target_sources(lokinet-amalgum PRIVATE
+  add_library(lokinet-hive-tooling
+    STATIC
     tooling/router_hive.cpp
     tooling/hive_router.cpp
     tooling/hive_context.cpp
   )
+  target_link_libraries(lokinet-tooling INTERFACE lokinet-hive-tooling)
+  target_link_libraries(lokinet-hive-tooling PUBLIC lokinet-context)
 endif()
 
-# TODO: make libunbound hidden behind a feature flag like sqlite for embedded lokinet
-target_link_libraries(lokinet-amalgum PRIVATE libunbound)
+# interface libraries for internal linkage
+add_library(lokinet-layers INTERFACE)
+add_library(lokinet-amalgum INTERFACE)
+
+# internal public linkages of components (order me)
+
+target_link_libraries(lokinet-config PUBLIC lokinet-util lokinet-nodedb lokinet-dns lokinet-platform)
+
+target_link_libraries(lokinet-context PUBLIC
+  lokinet-cryptography
+  lokinet-platform
+  lokinet-config
+  lokinet-peerstats
+  lokinet-layers
+  lokinet-consensus)
+
+target_link_libraries(lokinet-consensus PUBLIC lokinet-util)
+target_link_libraries(lokinet-consensus PRIVATE lokinet-context)
```

Comment:
You can combine these PUBLIC/PRIVATE linking into the same command which makes it a bit easier to read:
```cmake
target_link_libraries(lokinet-consensus
    PUBLIC lokinet-util
    PRIVATE lokinet-context)
```